cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/PolarizationState.java

```diff
@@ -0,0 +1,110 @@
+ *  indirect, special or general,        dommage, direct ou indirect,
+ *  consequential or incidental,         particulier ou général,
+ *  arising from the use of the          accessoire ou fortuit, résultant
+ *  software.  Neither the name          de l'utilisation du logiciel. Ni
+ *  of the National Research             le nom du Conseil National de
+ *  Council of Canada nor the            Recherches du Canada ni les noms
+ *  names of its contributors may        de ses  participants ne peuvent
+ *  be used to endorse or promote        être utilisés pour approuver ou
+ *  products derived from this           promouvoir les produits dérivés
+ *  software without specific prior      de ce logiciel sans autorisation
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fits.slice;
+
+/**
+ * Represents the various Polarization state values to be used when calculating cutouts.
+ */
+public enum PolarizationState {
```

Comment:
This is from caom2 but a good improvement from the 2011 code to use the enum mechanics to encode the names-number mapping :-)

For general use, this class should be up in cadc-dali (main dali pkg is fine). And it needs to have a static methods for both int -> PolarizationState and String -> PolarizationState. The latter will be needed when I remove the class from caom2 and it can be used in CommonParamValidator.validatePOL to validate input. Early validation of input is always a win.

That validatePOL method signature could be changed to return List<PolarizationState>, which would in turn change the type in the soda Cutout class... that will continue to trickle down into other code so it's a bit of a pain. But String all through a set of APIs usually leads to obscure bugs. Maybe leave that change for last and  see how painful it might be... problem with String is that it gets used in ways that the compiler won't detect (because of automagic toString() but also equals).

Reply:
For a String -> PolarizationState, is this the same as `Enum.valueOf(name)`?

Reply:
Could be, some enums in caom2 have values that are not valid enum names ("X-ray") hence the explicit method that loops over values():
```
    public static PolarizationState toValue(String s) {
        for (PolarizationState ps : values()) {
            if (ps.value.equals(s)) {
                return ps;
            }
        }
        throw new IllegalArgumentException("invalid value: " + s);
    }
```
which also gives you control over the exception and message. You would have to change the check to use the name since value was the String in the above snippet.